tests/dogtag/pytest-ansible/pytest/performance_test/test_cert_revocation.py

```diff
@@ -0,0 +1,79 @@
+"""
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Description: Load Test client Server Side configuration
+#   Steps :-
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+#
+#   Author: Deepak Punia <dpunia@redhat.com>
+#
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Copyright Red Hat, Inc.
+# SPDX-License-Identifier: GPL-2.0-or-later
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+"""
+import re
+import sys
+import logging
+import threading
+import argparse
+from pki.client import PKIConnection
+from pki.cert import CertClient
+
+parser = argparse.ArgumentParser()
+parser.add_argument("--hostname", help="<CA hostname>")
+parser.add_argument("--port", help="<CA Port Number>")
+parser.add_argument("--client-cert", help="path for admin.pem certificate")
+parser.add_argument("--number-of-clients", help="Number of thread", type=int)
+parser.add_argument("--cert_sn_file", help="path for certificate serial number file")
+
+log = logging.getLogger()
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+
+args = parser.parse_args()
+
+# Create a PKIConnection object that stores the details of the CA.
+connection = PKIConnection('https', args.hostname, args.port)
+
+# The pem file used for authentication. Created from a p12 file using the
+# command -
+# openssl pkcs12 -in <p12_file_path> -out /tmp/auth.pem -nodes
+connection.set_authentication_cert(args.client_cert)
+
+# Instantiate the CertClient
+cert_client = CertClient(connection)
+
+
+def run_test(cert_sn_file):
+    # execute the specified number of tests
+    cert_sn = cert_sn_file.readlines()
+    for cert_sn in cert_sn:
+        try:
+            cert_sn = re.findall('Serial Number:.*', cert_sn)[0].split(":")[1].strip()
+            revoke_data = cert_client.revoke_cert(cert_sn, revocation_reason='Key_Compromise')
+            if revoke_data.operation_result == 'success':
+                log.info("Revoke ID: {}".format(revoke_data.request_id))
+        except Exception as IndexError:
+            log.info(IndexError)
+
+
+if __name__ == "__main__":
+    # get test parameters from CLI parameters
+    number_of_threads = args.number_of_clients
+
+    # execute the test_cert_enrollement.py script and store Serial Number data into file
+    # provide that file path to this script by using --cert_sn_file <file path>
+    cert_sn_file = open(args.cert_sn_file, 'r')
```

Comment:
I'm not sure if `cert_sn_file.readlines()` is thread-safe. If it's a problem, you could read the whole file into a list, then split the list among the threads.

Reply:
It isn't thread safe as the first one would consume data and the rest would get empty data. @dpuniaredhat should probably pass `cert_sn_file` to `run_test` and call `open` in each.

Reply:
Changed the logic and tested the same to revoke 2M certificate and it's working fine.

Reply:
See my other comment. I think we should load the file into a list in the main thread, then split the list into smaller lists and assign them to each thread.

Reply:
Again, I think this'd be very easy with `json` and a `map`'d ThreadPool executor. My 2c. tho.